src/base/bittorrent/session.cpp

```diff
@@ -2753,6 +2761,59 @@ QStringList Session::bannedIPs() const
     return m_bannedIPs;
 }
 
+#if defined(Q_OS_WIN)
+OSMemoryPriority Session::getOSMemoryPriority() const
+{
+    return m_OSMemoryPriority;
+}
+
+void Session::setOSMemoryPriority(const OSMemoryPriority priority)
+{
+    if (m_OSMemoryPriority == priority)
+        return;
+
+    m_OSMemoryPriority = priority;
+    configureDeferred();
+}
+
+void Session::applyOSMemoryPriority() const
+{
+    using SETPROCESSINFORMATION = BOOL (WINAPI *)(HANDLE, PROCESS_INFORMATION_CLASS, LPVOID, DWORD);
+    const auto setProcessInformation = Utils::Misc::loadWinAPI<SETPROCESSINFORMATION>("Kernel32.dll", "SetProcessInformation");
+    if (!setProcessInformation)  // only available on Windows >= 8
+        return;
+
+#if (WIN32_WINNT < _WIN32_WINNT_WIN8)
+    // this dummy struct is required to compile successfully when targeting older Windows version
```

Comment:
Why not just exclude all the code below in this case? It will not executed anyway.

Reply:
>I don't get what you mean

```c++
#if (WIN32_WINNT >= _WIN32_WINNT_WIN8)
    MEMORY_PRIORITY_INFORMATION prioInfo {};
		    switch (getOSMemoryPriority()) {
		    case OSMemoryPriority::Normal:
		    default:
		        prioInfo.MemoryPriority = MEMORY_PRIORITY_NORMAL;
		        break;
		    case OSMemoryPriority::BelowNormal:
		        prioInfo.MemoryPriority = MEMORY_PRIORITY_BELOW_NORMAL;
		        break;
		    case OSMemoryPriority::Medium:
		        prioInfo.MemoryPriority = MEMORY_PRIORITY_MEDIUM;
		        break;
		    case OSMemoryPriority::Low:
		        prioInfo.MemoryPriority = MEMORY_PRIORITY_LOW;
		        break;
		    case OSMemoryPriority::VeryLow:
		        prioInfo.MemoryPriority = MEMORY_PRIORITY_VERY_LOW;
		        break;
		    }
		    setProcessInformation(::GetCurrentProcess(), ProcessMemoryPriority, &prioInfo, sizeof(prioInfo));
#endif
```

Reply:
That won't work because https://github.com/qbittorrent/qBittorrent/blob/master/winconf.pri#L13 is equal to `_WIN32_WINNT_WIN7` and the binary won't contain these code that should execute on Windows >= 8.

Reply:
Oops... I misinterpreted it.

But why MEMORY_PRIORITY_INFORMATION isn't available? Is it included conditionally?
MEMORY_PRIORITY_\* constants are still available...

Reply:
>Is it included conditionally?

Yes.

Reply:
It's weird...